src/many/src/main.rs

```diff
@@ -591,9 +592,12 @@ async fn main() {
                         }
                         .into_iter()
                         .collect(),
-                    )
-                    .build()
-                    .unwrap();
+                    );
+                if let Some(ts) = timestamp {
+                    builder.timestamp(Timestamp::from_system_time(ts).unwrap());
+                }
```

Comment:
Assuming this is the bug fix, but not relevant to the replay issue since this is only when a server argument is not passed